clients/admin-ui/src/features/privacy-requests/RequestTable.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useRef, useState } from 'react';
+import {
+  Table,
+  Text,
+  Thead,
+  Tbody,
+  Tr,
+  Th,
+  Button,
+  Flex,
+} from '@fidesui/react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import debounce from 'lodash.debounce';
+
+import { PrivacyRequest } from './types';
+import {
+  selectPrivacyRequestFilters,
+  useGetAllPrivacyRequestsQuery,
+  setPage,
+} from './privacy-requests.slice';
+
+import RequestRow from './RequestRow';
+
+interface RequestTableProps {
+  requests?: PrivacyRequest[];
+}
+
+const useRequestTable = () => {
+  const dispatch = useDispatch();
+  const filters = useSelector(selectPrivacyRequestFilters);
+  const [cachedFilters, setCachedFilters] = useState(filters);
+  const updateCachedFilters = useRef(
+    debounce((updatedFilters) => setCachedFilters(updatedFilters), 250)
+  );
+  useEffect(() => {
+    updateCachedFilters.current(filters);
+  }, [setCachedFilters, filters]);
+
+  const handlePreviousPage = () => {
+    dispatch(setPage(filters.page - 1));
```

Comment:
Am I understanding correctly that this action is for pagination on the table? And am I missing the limit on the number of requests displayed per page?

Reply:
Also, maybe this is more a question for design, but I'm wondering if it's worth it to list the total results returned for a search

Reply:
Yes, that sets the current page. The `size` param is stored in the `subjectRequests` slice state (initialized to 25 [here](https://github.com/ethyca/fidesops/pull/364/files#diff-8304db3d5ee974fc8bf6614b0d59aea2962e5919d4bbf411cd0f8228d12eb70aR148)). Eventually we'll probably want to add a dropdown to change size, which would dispatch the `setSize` action, but for the time being it's just static.

And we do list the total results returned for a search [`here`](https://github.com/ethyca/fidesops/pull/364/files#diff-56dcf8c8d01510145ff6436a98f83f0dde98131cee06cf737e702846bef5830cR86):

<img width="340" alt="image" src="https://user-images.githubusercontent.com/6969740/162234075-c3834755-fe30-4c52-a5b3-619a4ee2626d.png">